lib/rubocop/cop/style/yoda_condition.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for Yoda conditions, i.e. comparison operations where
+      # readability is reduced because the operands are not ordered the same
+      # way as they would be ordered in spoken English.
+      #
+      # @example
+      #
+      #   # bad
+      #   99 == foo
+      #   "bar" == foo
+      #
+      # @example
+      #
+      #   # good
+      #   foo == 99
+      #   foo == "bar"
+      class YodaCondition < Cop
+        MSG = 'Reverse the order in conditional statement: `%s` == `%s`.'.freeze
+        WHITELIST_TYPES = %i(lvar ivar send const regexp begin).freeze
+
+        def on_send(node)
+          return unless yoda_condition?(node)
+
+          register_offense(node)
+        end
+
+        private
+
+        def comparison_operator?(node)
+          RuboCop::AST::Node::COMPARISON_OPERATORS.include?(node.method_name)
+        end
+
+        def yoda_condition?(node)
```

Comment:
@rrosenblum Thanks for comment. Where I can read more about NodePattern? Do you have some example on top of your mind?